app/stac_api/views.py

```diff
@@ -106,6 +112,92 @@ def parse_datetime_query(date_time):
     return start, end
 
 
+def filter_by_bbox(queryset, bbox):
+    '''Filter a querystring with a given bbox
+
+    This function is a helper function, for some views, to add a bbox filter to the queryset.
+
+    Args:
+        queryset:
+            A django queryset (https://docs.djangoproject.com/en/3.0/ref/models/querysets/)
+        bbox:
+            A string defining a spatial bbox (f.ex. 5.96, 45.82, 10.49, 47.81)
+
+    Returns:
+        The queryset with the added spatial filter
+
+    Raises:
+        ValidationError: When the bbox does not contain 4 values. Or when the polygon build
+        from the bbox string is invalid.
+    '''
+    try:
+        logger.debug('Item query parameter bbox = %s', bbox)
+        query_bbox_polygon = Polygon.from_bbox(bbox.split(','))
+    except ValueError as error:
+        logger.error(
+            'Invalid bbox parameter: '
+            'Could not transform bbox "%s" to a polygon; %s'
+            'f.ex. bbox=5.96, 45.82, 10.49, 47.81',
+            bbox,
+            error
+        )
+        raise ValidationError(_('Invalid bbox query parameter, '
+                                ' has to contain 4 values. f.ex. bbox=5.96,45.82,10.49,47.81'))
+    query_bbox_polygon.srid = 4326
+    validate_geometry(query_bbox_polygon)
+    return queryset.filter(geometry__intersects=query_bbox_polygon)
+
+
+def filter_by_datetime(queryset, date_time):
+    '''Filter a queryset by datetime
+
+    Args:
+        queryset:
+             A django queryset (https://docs.djangoproject.com/en/3.0/ref/models/querysets/)
+        date_time:
+            A string
+    Returns:
+        The queryset filtered by date_time
+    '''
+    start, end = parse_datetime_query(date_time)
+    if end is not None:
+        return _filter_by_datetime_range(queryset, start, end)
+    return queryset.filter(properties_datetime=start)
+
+
+def _filter_by_datetime_range(queryset, start_datetime, end_datetime):
+    '''Filter a queryset by datetime range
+
+    Helper function of filter_by_datetime
```

Comment:
FOR NEXT PR: dito

Reply:
https://github.com/geoadmin/service-stac/pull/175